aarch64/src/l.S

```diff
@@ -0,0 +1,29 @@
+// Aarch64 entry (Raspberry Pi 3, 4 focussed)
+ 
+.section .boottext, "awx"
+.globl start
+
+start:
+	// All cores other than 0 should just hang
+    mrs     x1, mpidr_el1
+    and     x1, x1, #3
+    cbnz    x1, 5f
+	
+	// On the main core, so set up the initial stack
+	ldr	x5, =start
+	add	x5, x5, #4096
+	mov	sp, x5
+
+	// Clean the BSS section
+    ldr     x1, =bss     		// Start address
+    ldr     w2, =ebss  		    // Size of the section
+3:  cbz     w2, 4f              // Quit loop if zero
+    str     xzr, [x1], #8
```

Comment:
This doesn't seem right in several dimensions:
* Where do you increment `x1`? Doesn't this loop just store 8-bytes worth of 0's into *bss however-many times?
* Why use `w2` for ebss? What if the BSS is particularly large? Use an x reg.
* You never subtract the start of the BSS from w2, so w2 is an address, not a length.
* Since `bss` isn't necessarily aligned, `x1` might point to an unaligned address, so `str` here seems wrong.
* You're storing words, but decrementing w2 by 1, which implies storing bytes. You shouldn't probably either store bytes or decrement the count by the word length.

Reply:
Ah, I see the post-increment addressing mode on `str` so `x1` is incremented, but I think the rest of the feedback holds.

Reply:
tbh a lot of this was patched together from other sources.  rewrote a lot of this, should be a bit more sensible now.